src/network/macvlan_dhcp.rs

```diff
@@ -75,6 +75,25 @@ pub fn get_dhcp_lease(
         None
     };
 
+    let dns_servers = if !lease.dns_servers.is_empty() {
+        let servers = lease
+            .dns_servers
+            .into_iter()
+            .map(|d| {
+                match IpAddr::from_str(&d) {
+                    Ok(d) => Ok(d),
+                    Err(e) => {
+                        return Err(NetavarkError::msg(format!("bad dns address: {e}")));
+                    }
+                }
+            })
+            .collect::<Result<Vec<IpAddr>, NetavarkError>>()?;
+        Some(servers)
+    } else {
+        None
+    };
+    let domain_name = Some(lease.domain_name).filter(|s| !s.is_empty());
```

Comment:
It seems better to do something like this
```
let domain_name = if !lease.domain_name.is_empty() {
        Some(vec![lease.domain_name])
    } else {
        None
    };
```
Then you do not need to convert in the caller side again.